src/optimizer/transforms/char-class-classranges-merge-transform.js

```diff
@@ -95,10 +95,13 @@ function sortCharClass(a, b) {
 function getSortValue(expression) {
   if (expression.type === 'Char') {
     if (expression.value === '-') {
-      return Infinity;
+      return Number.MAX_SAFE_INTEGER;
+    }
+    if (expression.value === '^') {
+      return Number.MAX_SAFE_INTEGER - 1;
```

Comment:
Should it also be `Number.MAX_SAFE_INTEGER`? In this case `/[-^]/` would stay the same.